db/range_del_aggregator.cc

```diff
@@ -587,32 +585,60 @@ void RangeDelAggregator::InvalidateRangeDelMapPositions() {
     return;
   }
   for (auto& stripe : rep_->stripe_map_) {
-    stripe.second->InvalidatePosition();
+    stripe.second.first->InvalidatePosition();
   }
 }
 
-RangeDelMap& RangeDelAggregator::GetRangeDelMap(SequenceNumber seq) {
+RangeDelMap* RangeDelAggregator::GetRangeDelMapIfExists(SequenceNumber seq) {
   assert(rep_ != nullptr);
   // The stripe includes seqnum for the snapshot above and excludes seqnum for
   // the snapshot below.
+  if (rep_->stripe_map_.empty()) {
+    return nullptr;
+  }
   StripeMap::iterator iter;
   if (seq > 0) {
-    // upper_bound() checks strict inequality so need to subtract one
-    iter = rep_->stripe_map_.upper_bound(seq - 1);
+    iter = rep_->stripe_map_.lower_bound(seq);
   } else {
     iter = rep_->stripe_map_.begin();
   }
+  if (iter == rep_->stripe_map_.end()) {
+    return nullptr;
+  }
+  size_t snapshot_idx = iter->second.second;
+  if (snapshot_idx > 0 && seq <= rep_->snapshots_[snapshot_idx - 1]) {
+    return nullptr;
+  }
+  return iter->second.first.get();
+}
+
+RangeDelMap& RangeDelAggregator::GetRangeDelMap(SequenceNumber seq) {
+  assert(rep_ != nullptr);
+  // The stripe includes seqnum for the snapshot above and excludes seqnum for
+  // the snapshot below.
+  std::vector<SequenceNumber>::iterator iter;
+  if (seq > 0) {
+    iter = std::lower_bound(rep_->snapshots_.begin(), rep_->snapshots_.end(),
+                            seq);
+  } else {
+    iter = rep_->snapshots_.begin();
```

Comment:
I'd expect lower_bound to pick the begin element when `seq == 0` since `snapshots_` shouldn't hold anything negative.